pkg/git/repository.go

```diff
@@ -105,6 +108,7 @@ func AddFlags(cmd *cobra.Command, bfs billy.Filesystem, prefix string) *CloneOpt
 
 		envPrefix := strings.ReplaceAll(strings.ToUpper(prefix), "-", "_")
 		cmd.PersistentFlags().StringVar(&co.Auth.Password, prefix+"git-token", "", fmt.Sprintf("Your git provider api token [%sGIT_TOKEN]", envPrefix))
+		cmd.PersistentFlags().StringVar(&co.Provider, prefix+"provider", "", fmt.Sprintf("The git provider, one of: %v", strings.Join(Providers(), "|")))
```

Comment:
This flag can be added only in cases where the repository would be created, in all other cases this flag is just confusing. Lets try to add it only to commands that can create a repo.